code/clue.js

```diff
@@ -1,59 +1,342 @@
+  lounge, 
+  ballRoom,
+  hall,
+  spa,
+  livingRoom,
+  observatory, 
+  theater, 
+  guestHouse, 
+  patio
+];
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
-  return array[Math.floor(Math.random() * array.length)]
-}
+  return array[Math.floor(Math.random() * array.length)];
+};
+
+// CREATE AN OBJECT THAT KEEPS THE MYSTERY. With a killer, a weapon and a room. The values are set later down.
+let theMystery = { // Write mystery instead?
+ killer: '',
+ weapon: '',
+ room: ''
+};
+
+// Variables for the pictures 
+const killerImage = document.getElementById('killerImage');
+const weaponImage = document.getElementById('weaponImage');
+const roomImage = document.getElementById('roomImage');
+
+// PICK-KILLER-FUNCTION! This function will be invoked when you click on the killer card.
+const pickKiller = () => { // Shows loading animation
+  document.getElementById('loader1').style.display = 'block';
+  setTimeout(realPickKiller, 1000); // Delays the realPickKiller function and calls it after 1.5 sec.
+};
+document.getElementById('killerCard').addEventListener('click', pickKiller);
+
+const realPickKiller = () => { // The REAL function getting the values for the suspects
+  document.getElementById('loader1').style.display = 'none';
+  document.getElementById('killerCard').removeEventListener('click', pickKiller); // So the loader doesn't pops up in next click
+ 
+  mystery.killer = randomSelector(suspects);  // This will randomly select a killer from the suspects and add that to the mystery object.
+
+  // This function does randomly select a favourite weapon.
+  shuffleFavouriteWeapon = () => {
+    mystery.favouriteWeapon = randomSelector(weapons); 
+  };
+  shuffleFavouriteWeapon();
+
+  // This will change the background color of the card to the one connected to the chosen killer and show the different properties of the killer. 
+  document.getElementById('killerCard').style.background = mystery.killer.color;
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  document.getElementById('killerAge').innerHTML = `Age: ${mystery.killer.age}`;
+  document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`;
+  killerImage.src = mystery.killer.image;
+  document.getElementById('killerDescription').innerHTML = `Description: ${mystery.killer.description}`;
+  document.getElementById('favouriteWeapon').innerHTML = `Favourite weapon: ${mystery.favouriteWeapon.name}`;
+
+  document.getElementById('killerCard').onclick = realPickKiller;
+};
+
+// PICK-WEAPON-FUNCTION! This function will be invoked when you click on the weapon card.- in the similar way as pickKiller.
+const pickWeapon = () => { // Shows loading animation
+  document.getElementById('loader2').style.display = 'block';
```

Comment:
`loader2` seems like it could have a different name to provide more context.  Maybe something like `weaponLoader`